kong/plugins/moesif/log.lua

```diff
@@ -241,10 +246,28 @@ local function send_events_batch(premature)
         end
         if type(send_events_socket) == "table" and next(send_events_socket) ~= nil then
           local counter = 0
+          local total_events_sent_in_this_cycle = 0
           repeat
             local event = table.remove(queue)
             counter = counter + 1
             table.insert(batch_events, event)
+            -- Add total number of events to send in this cycle
+            total_events_sent_in_this_cycle = total_events_sent_in_this_cycle + 1
+
+            if (total_events_sent_in_this_cycle == configuration.max_events_sent_per_callback) then 
+              local start_pay_time = socket.gettime()*1000
+               if pcall(send_payload, send_events_socket, parsed_url, batch_events, configuration) then 
+                sent_event = sent_event + #batch_events
+               end
+               local end_pay_time = socket.gettime()*1000
+               if configuration.debug then
+                ngx_log(ngx.DEBUG, "[moesif] send payload with event count - " .. tostring(#batch_events) .. " took time - ".. tostring(end_pay_time - start_pay_time).." for pid - ".. ngx.worker.pid())
+               end
+               batch_events = {}
+               ngx_log(ngx.DEBUG, "[moesif] Max events sent per callback, skip sending more events - " .." for pid - ".. ngx.worker.pid())
+               break
```

Comment:
this break only breaks this loop.. So at line 238, check that total_events_sent_in_this_cycle< configuration.max_events_sent_per_callback